[TEST] Add Attested TLS Test Case

triston996 opened the pull request:
Add attested tls test case.

**Note** This PR will not merge before @jxyang merge his PR to OE sdk. Because we need his code change to verify type 1 cert.

This test case will **run two enclaves**: oe_enclave and sgxlkl_enclave. The oe_enclave will run as tls server and the sgxlkl_enclave will be the tls client. 
Since the sgxlkl_enclave is not able to generate the certificate now, this test case will only cover the tls server verification part by using pre-generated cert (type 1). We will cover that part once sgxlkl_enclave support the cert generation.

**Test**:
1. Manual test by run oe_encalve and sgxlkl_enclave separately.
2. Tested by using `run_scenario.sh`
Test Result Log: 
`Tue Jul 21 05:54:14 UTC 2020

 Server in enclave: Waiting for a trusted connection
Server: Received TLS certificate.
  ccrt->version = 3 cert_size = 5069
Server: attestation certificate verified.

=== _verify_identity()

 Remote connection established. Ready for service.
[    0.575887] reboot: Restarting system
TEST PASSED`

triston996 committed:
add attested tls test case

triston996 committed:
add attested tls test case

triston996 committed:
add attested tls test case

triston996 committed:
add attested tls test case

triston996 committed:
add attested tls test case

triston996 commented:
@hukoyu @Pengpeng-Microsoft @jxyang 
I can't add reviewer now, please take a look and let me know who else should I notice for review purpose.

letmaik commented:
If this is only testing whether attested TLS works, why does the client have to be in an enclave? Couldn't the test be simplified a lot by using a non-enclave client?

hukoyu commented on tests/attested_tls/run_scenario.sh:
```diff
@@ -0,0 +1,82 @@
+fi
+
+# Install Open Enclave if not installed
+if [[ ! -d "/opt/openenclave" && $cert_pr_merged_to_oe -eq 1 ]]; then
```
You don't need "cert_pr_merged_to_oe -eq 1"  part.

hukoyu commented on tests/attested_tls/run_scenario.sh:
```diff
@@ -0,0 +1,82 @@
+    sleep 5
+fi
+
+process="client_host"
```
I think process must be tlssrv_host

hukoyu reviewed (changes requested):
I gave some feedback for run_scenario.sh

hukoyu commented:
@triston996  build is failing due to check-ci errors on run_scenario.sh
Run scripts/check-ci from source root folder and fix all errors on run_scenario.sh otherwise build fails and tests don't run.

triston996 committed:
add attested tls test case

triston996 committed:
add attested tls test case

triston996 committed:
fix ci check

triston996 committed:
fix ci check

triston996 commented:
This task is to cover the customer using scenario where SGX-LKL based enclave and OE SDK based enclave establish attested TLS communication channel between. ( cc: @Pengpeng-Microsoft, Please correct me if I understand incorrectly.) 
So, @letmaik thanks for the feedback and I understand your concern, but I think the intention here is to create two enclaves and test the tls connection between. I copied Pengpeng to verify the idea and we can setup a discussion if you have more concerns.

Pengpeng-Microsoft commented on tests/attested_tls/common/gencreds.cpp:
```diff
@@ -0,0 +1,272 @@
+// Copyright Microsoft. 
```
I did not find reference to the function generate_attested_credentials. Do you need this file?

triston996 replied:
This func is been used in "_configure_srv" in oe_enclave tls server. Do you mean, I should use the certificate which mbedtls provided?

Pengpeng-Microsoft commented on tests/attested_tls/common/settings.c:
```diff
@@ -0,0 +1,55 @@
+        goto fail;
+    }
+
+    if (getenv(name) == NULL)
```
Can you remove the whole if block here? This would pause the pipeline if the env variables do not exist.

triston996 replied:
I think there is a timeout value, but it make sense to me to remove the entire if block.

Pengpeng-Microsoft commented on tests/attested_tls/oe_enclave/enc/private.pem:
```diff
@@ -0,0 +1,39 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Do you need this file?

triston996 replied:
this is for the "oesign", I think we need to keep it here. But please correct me if I understand incorrectly.

hukoyu replied:
Verified with Xuejun, this file is needed for a known MRSIGNER on sgx-lkl enclave side.

Pengpeng-Microsoft commented on tests/attested_tls/oe_enclave/enc/public.pem:
```diff
@@ -0,0 +1,39 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Do you need this file?

triston996 replied:
please check the above one

triston996 replied:
Confirmed w/ Xuejun, public.pem can be removed.

Pengpeng-Microsoft commented on tests/attested_tls/oe_enclave/enc/server.cpp:
```diff
@@ -0,0 +1,829 @@
+// Copyright Microsoft. 
```
Can you rename the file to "tlssrv.cpp"?

triston996 replied:
sure, I will change it.

Pengpeng-Microsoft reviewed (changes requested):
Can you also remove file .Makefile.swp ?

Pengpeng-Microsoft commented:
> This task is to cover the customer using scenario where SGX-LKL based enclave and OE SDK based enclave establish attested TLS communication channel between. ( cc: @Pengpeng-Microsoft, Please correct me if I understand incorrectly.)
> So, @letmaik thanks for the feedback and I understand your concern, but I think the intention here is to create two enclaves and test the tls connection between. I copied Pengpeng to verify the idea and we can setup a discussion if you have more concerns.

Yes, the intention is to test the attested tls connection scenario between SGX-LKL and OE SDK based enclaves.

triston996 committed:
code clean up

triston996 committed:
code clean up

hukoyu commented on tests/attested_tls/common/settings.c:
```diff
@@ -0,0 +1,24 @@
+#include "settings.h"
```
You can remove settings.c file. 
In the referred files use "#include <stdlib.h> and getenv() function directly.

hukoyu commented on tests/attested_tls/common/settings.h:
```diff
@@ -0,0 +1,15 @@
+#ifndef COMMON_SETTINGs_H
```
Same comment above. Remove settings.h

hukoyu commented on tests/attested_tls/oe_enclave/makefile:
```diff
@@ -0,0 +1,18 @@
+all: build
+
+build:
+	@. /opt/openenclave/share/openenclave/openenclaverc
```
Please remove this line. This must be called in run_scenario.sh

hukoyu commented on tests/attested_tls/oe_enclave/makefile:
```diff
@@ -0,0 +1,18 @@
+	$(MAKE) -C host clean
+
+run:
+	@. /opt/openenclave/share/openenclave/openenclaverc
```
Please remove this line. This must be called in run_scenario.sh

hukoyu commented on tests/attested_tls/run_scenario.sh:
```diff
@@ -0,0 +1,78 @@
+    sudo apt -y install clang-7 libssl-dev gdb libsgx-enclave-common libsgx-enclave-common-dev libprotobuf10 libsgx-dcap-ql libsgx-dcap-ql-dev az-dcap-client open-enclave
+fi
+
+# Set log file that will be shared between two enclaves in the test scenario for stdout and stderr logs
```
Add here
"# shellcheck disable=SC1091"
". /opt/openenclave/share/openenclave/openenclaverc"

hukoyu commented on tests/attested_tls/run_scenario.sh:
```diff
@@ -0,0 +1,78 @@
+# oe_enclave is the tls server and runs and waits for connection from sgxlkl_enclave.
+# then sgxlkl_enclave starts and connects to oe_enclave by using pre-created sgxlkl_cert.der and sgxlkl_private_key.pem
+# then both sides will shows the result
+
```
Add here:
"# Run mode is sw or hw"
"export SGXLKL_RUN_MODE=$1"

hukoyu commented on tests/attested_tls/Makefile:
```diff
@@ -0,0 +1,21 @@
+all: run-hw run-sw
+
+run-sw:
+	@export SGXLKL_RUN_MODE=sw
```
Remove export line

hukoyu commented on tests/attested_tls/Makefile:
```diff
@@ -0,0 +1,21 @@
+
+run-sw:
+	@export SGXLKL_RUN_MODE=sw
+	@bash run_scenario.sh
```
Make this 
"@bash run_scenario.sh sw"

hukoyu commented on tests/attested_tls/Makefile:
```diff
@@ -0,0 +1,21 @@
+	@bash run_scenario.sh
+
+run-hw:
+	@export SGXLKL_RUN_MODE=hw
```
Remove export line

hukoyu commented on tests/attested_tls/Makefile:
```diff
@@ -0,0 +1,21 @@
+
+run-hw:
+	@export SGXLKL_RUN_MODE=hw
+	@bash run_scenario.sh
```
Make this:
"@bash run_scenario.sh hw"

hukoyu commented on tests/attested_tls/run_scenario.sh:
```diff
@@ -0,0 +1,78 @@
+# Build and run OE enclave app
+cd oe_enclave || exit 1
+make
+make run
```
Before "make run", check if build succeeded. Add this code
exit_code=$?
if [[ $exit_code -ne 0 ]]; then
    echo "Failed to build OE_ENCLAVE"
    exit 1
fi

hukoyu commented on tests/attested_tls/run_scenario.sh:
```diff
@@ -0,0 +1,78 @@
+cd ..
+cd sgxlkl_enclave || exit 1
+make lkl-image
+make lkl-run
```
Before "make lkl-run" check if built successfully:
exit_code=$?
if [[ $exit_code -ne 0 ]]; then
    echo "Failed to build SGXLKL_ENCLAVE"
    exit 1
fi

hukoyu reviewed (changes requested):
@triston996 while working on MAA remote attestation PR, I fixed several issues. I provided you feedback those issues in your PR as well.

hukoyu commented on tests/attested_tls/oe_enclave/host/makefile:
```diff
@@ -0,0 +1,38 @@
+	$(CC) -c $(CFLAGS) $(INCLUDES) tlssrv_u.c
+	$(CC) -c $(CXXFLAGS) $(INCLUDES) $(COMMONS)/settings.c
+	$(CXX) -c $(CXXFLAGS) $(INCLUDES) host.cpp
+	$(CXX) -o tlssrv_host host.o tlssrv_u.o settings.o $(LDFLAGS) -lpthread -lcurl -lmbedtls -lmbedcrypto
```
You need to add -L/opt/openenclave/lib/openenclave/enclave after $(LDFLAGS) , otherwise you get a build error saying cannot find lmbedtls and lmbedcrypto

jxyang commented on .azure-pipelines/other/nightly_run_only_tests.txt:
```diff
@@ -1 +1,2 @@
 tests/ml/openvino/Makefile
+tests/attested_tls/Makefile
```
This is related to Huseyin's test case. Suggest both test cases to be placed under `tests/attestation`. This one named `sgxlkl_attests_to_oe`.

jxyang commented on tests/attested_tls/Makefile:
```diff
@@ -0,0 +1,21 @@
+# This test case will run two enclaves: oe_enclave and sgxlkl_enclave
```
Prefer the present tense, e.g., `This test case runs two enclaves...`. This applies to other descriptions.

jxyang commented on tests/attested_tls/Makefile:
```diff
@@ -0,0 +1,21 @@
+# sgxlkl_enclave should be able to generate the certificate and send to tls server for verify
+# however, sgxlkl_enclave not able to gernerate the cert now.
+# So this test case will only cover the tls server cert verification part
+# which use the pregenerated cert (type 1)
```
Type 1 is an informal term that you can omit.

triston996 replied:
do I need to mention type 1 in other place? are we going to update the type# nearly?

jxyang commented on tests/attested_tls/common/gencreds.cpp:
```diff
@@ -0,0 +1,269 @@
+#include "gencreds.h"
```
As I commented in Huseyin's PR, this file should be under `tests/attestation/common`, and be shared by both Huseyin's and your test cases.

triston996 replied:
I think this make sense to me, but maybe we should have a follow up PR to change Huseyin's RP and this PR together to reduce the duplicated work

jxyang commented on tests/attested_tls/oe_enclave/enc/makefile:
```diff
@@ -0,0 +1,45 @@
+# Copyright Microsoft. 
+# Licensed under the attached Microsoft Software License Terms
```
Remove the copyright lines. It's not applicable to sgx-lkl.

jxyang commented on tests/attested_tls/oe_enclave/enc/makefile:
```diff
@@ -0,0 +1,45 @@
+# Licensed under the attached Microsoft Software License Terms
+
+# Detect C and C++ compiler options
+# if not gcc and g++, default to clang-7
```
SGX-LKL uses gcc. You can remove the unnecessary checking.

jxyang commented on tests/attested_tls/sgxlkl_enclave/tlscli.c:
```diff
@@ -0,0 +1,540 @@
+    return ret;
+}
+
+oe_result_t _verifier_callback(oe_identity_t* identity, void* arg)
```
This function is not used anywhere.

jxyang commented on tests/attested_tls/sgxlkl_enclave/oe_public/bits/result.h:
```diff
@@ -0,0 +1,375 @@
+
```
Why do we need this file?

triston996 replied:
to define _oe_result I think, do you think we need to remove it and define by ourself?

triston996 replied:
removed

jxyang commented on tests/attested_tls/sgxlkl_enclave/oe_public/bits/types.h:
```diff
@@ -0,0 +1,181 @@
+
```
Why do we need this file?

triston996 replied:
This is not needed, I just removed

triston996 replied:
Just verified, we don't need it

jxyang commented on tests/attested_tls/sgxlkl_enclave/oe_public/bits/report.h:
```diff
@@ -0,0 +1,208 @@
+
+/**
```
Why do we need this file?

triston996 replied:
Just verified, we don't need it

jxyang commented on tests/attested_tls/sgxlkl_enclave/oe_public/bits/defs.h:
```diff
@@ -0,0 +1,182 @@
+#ifndef _OE_BITS_DEFS_H
+#define _OE_BITS_DEFS_H
```
Why do we need this file?

triston996 replied:
this is used in result.h

triston996 replied:
removed

jxyang commented on tests/attested_tls/sgxlkl_enclave/makefile:
```diff
@@ -0,0 +1,30 @@
+# Copyright Microsoft. 
+# Licensed under the attached Microsoft Software License Terms
```
Remove the license terms.

jxyang commented on tests/attested_tls/sgxlkl_enclave/makefile:
```diff
@@ -0,0 +1,30 @@
+alpine-container:
+	docker build -t alpine3.10.build ./alpine
+
+alpine-binaries: alpine-container
```
Follow #664, try to remove  `alpine-container`, `alpine-binaries`,  the alpine folder.

jxyang commented on tests/attested_tls/sgxlkl_enclave/alpine/Dockerfile:
```diff
@@ -0,0 +1,22 @@
+# Copyright Microsoft. 
+# Licensed under the attached Microsoft Software License Terms
```
This can be removed if we compile the application in `sgxlkl_enclave/Dockerfile`, as #664  did.

jxyang commented on tests/attested_tls/sgxlkl_enclave/clientAgent.h:
```diff
@@ -0,0 +1,13 @@
+// The state object shared by server and client Agent
+typedef struct struct_clientAgentState {
+    mbedtls_net_context untrustedChannel;
+    tlscli_t* trustedChannel ;
```
We don't need this file if we declare `tlscli_t* trustedChannel` in `client.c`.

jxyang commented on tests/attested_tls/sgxlkl_enclave/client.c:
```diff
@@ -0,0 +1,92 @@
+  This is the entry point of sgxlkl_enclave which is responsible for:
+
+  1) Establishing a trusted channel with oe_enclave via clientAgent;
+  2) Verify the pregenerated cert (type = 1)
```
Remove `type == 1`.

jxyang commented on tests/attested_tls/sgxlkl_enclave/client.c:
```diff
@@ -0,0 +1,92 @@
+    int rc;
+    clientAgentState_t* state = calloc(1, sizeof(clientAgentState_t));
+
+    if (USE_UNTRUSTED_CHANNEL) {
```
Remove the untrusted channel implementation.

jxyang commented on tests/attested_tls/sgxlkl_enclave/client.c:
```diff
@@ -0,0 +1,92 @@
+#include <mbedtls/net_sockets.h>
+#include "tlscli.h"
+#include "clientAgent.h"
+#include "../common/settings.h"
```
Get rid of the dependency on settings.h and message.h, if there is any.

triston996 replied:
message.h I think we may keep it since the server_port is used in multiple files like, client.c, host.cpp. But for now, I can just copy the server_port and remove the message.h.

jxyang commented on tests/attested_tls/run_scenario.sh:
```diff
@@ -0,0 +1,78 @@
+#! /bin/bash
```
Please see my comments on this file from Huseyin's PR #650.

Also suggest putting this file under `tests/attestation/common` so it can shared by both test cases.

jxyang commented on tests/attested_tls/oe_enclave/tlssrv.edl:
```diff
@@ -0,0 +1,12 @@
+        public int setup_tls_server([in, string] const char* port);
+    };
+
+    untrusted {
```
I believe you can remove the empty `untrusted` block.

jxyang commented on tests/attested_tls/oe_enclave/makefile:
```diff
@@ -0,0 +1,18 @@
+# Copyright Microsoft. 
+# Licensed under the attached Microsoft Software License Terms.
```
Remove the license terms.

jxyang commented on tests/attested_tls/oe_enclave/host/host.cpp:
```diff
@@ -0,0 +1,54 @@
+
+    /* Check argument count */
+    if (argc != 2) {
+        printf("Usage: %s CLIENT_ENCLAVE_PATH\n", argv[0]);
```
This is the TLS server, not client.

jxyang commented on tests/attested_tls/oe_enclave/host/host.cpp:
```diff
@@ -0,0 +1,54 @@
+#include <pthread.h>
+#include <unistd.h>
+#include "tlssrv_u.h"
+#include "../../common/message.h"
```
Remove references to message.h, if there is any.

jxyang commented on tests/attested_tls/oe_enclave/enc/tlssrv.cpp:
```diff
@@ -0,0 +1,826 @@
+static mbedtls_net_context client_fd;
+static tlssrv_t* tlsServer = NULL;
+static tlssrv_err_t tlsError;
+uint8_t iv[16] = 
```
where is `iv` used?

triston996 replied:
removed

jxyang commented on tests/attested_tls/oe_enclave/enc/tlssrv.cpp:
```diff
@@ -0,0 +1,826 @@
+        goto done;
+
+    oe_result_t r;
+    if ((r = oe_verify_attestation_certificate(
```
If you use type 1 certificate from sgx-lkl side, then you need the new API introduced in my recent PR: [`oe_verify_attestation_certificate_with_evidence`](https://github.com/openenclave/openenclave/blob/37b2a1d0a41d93ca5c816f17ba23130c6e2f7831/tests/attestation_plugin_cert/README.md).

Alternatively, you could use type 0 certificate on sgx-lkl side, and continue using `oe_verify_attestation_certificate`.

Either way is fine with me. But remember our ultimate goal is to generate a type 2 certificate on SGX-LKL side, which contains EEID evidence. When the functionality is implemented in SGX-LKL, we have to switch to `oe_verify_attestation_certificate_with_evidence` here.

jxyang replied:
I wonder how could your test case passed the pipeline with this obvious mismatch between the certificate and the verifier. Could you double check the log to see if this test case is indeed run?

hukoyu replied:
@jxyang this test is marked for nightly run. So this test doesn't run when PR submitted.

triston996 replied:
@jxyang is that PR the same branch to generate the cert?

triston996 replied:
I switch back to type 0 cert because the current oe branch not support yet. I will update it once sgxlkl-support branch have all of the changes.

hukoyu commented:
@triston996  I implemented and verified skipping sw test mechanism. Please look at here : https://github.com/lsds/sgx-lkl/pull/650#issuecomment-667455518

cc @Pengpeng-Microsoft

triston996 committed:
skip sw mode

triston996 committed:
skip sw mode

triston996 committed:
fixed typo

triston996 committed:
rename the test folder

triston996 committed:
rename the test folder

triston996 committed:
removed unused files

triston996 committed:
removed unused files

triston996 committed:
updated log file location

triston996 committed:
updated log file location

triston996 commented:
@hukoyu @jxyang 
I addressed all of your comments. Since Huseyin's PR and this PR have some common code, we should have another follow up PR to reduce the duplicated works. Please take a look and let me know if that looks good to you.

triston996 committed:
updated script to install openenclave

hukoyu commented:
@triston996 can you share the link for sgx-lkl nightly pipeline run that shows this test passes for hw mode and skips for sw mode.

triston996 commented:
> @triston996 can you share the link for sgx-lkl nightly pipeline run that shows this test passes for hw mode and skips for sw mode.

@hukoyu I did the nightly pipeline test, entire pipeline fails but my test passed, and here is the result for my test log:
For HW: [link](https://tfsproduks1.visualstudio.com/Ab5352d9a-5ed5-48f2-b767-7f273b55b53c/ff25f828-9f87-48e4-94f0-7449609f7e8f/_apis/build/builds/1633/logs/412)
`2020-08-03T07:03:36.9204570Z Mon Aug  3 07:02:49 UTC 2020
2020-08-03T07:03:36.9204848Z Host: Enclave path = ./enc/tlssrv_enc.signed, Run mode = hw, server port = 13999
2020-08-03T07:03:36.9205105Z Host: starting_server
2020-08-03T07:03:36.9205575Z Azure Quote Provider: libdcap_quoteprov.so [ERROR]: Could not retreive environment variable for 'AZDCAP_DEBUG_LOG_LEVEL'
2020-08-03T07:03:36.9205853Z 
2020-08-03T07:03:36.9206123Z  Server in enclave: Waiting for a trusted connection
2020-08-03T07:03:36.9206373Z Server: Received TLS certificate.
2020-08-03T07:03:36.9206718Z   crt->version = 3 cert_size = 5066
2020-08-03T07:03:36.9206969Z Server: attestation certificate verified.
2020-08-03T07:03:36.9207129Z 
2020-08-03T07:03:36.9207299Z === _verify_identity()
2020-08-03T07:03:36.9207443Z 
2020-08-03T07:03:36.9207609Z  Remote connection established. Ready for service.
2020-08-03T07:03:36.9207842Z [    3.419409] reboot: Restarting system
2020-08-03T07:03:36.9208045Z TEST PASSED
2020-08-03T07:03:36.9208235Z Test run completed with EXIT CODE 0
2020-08-03T07:03:36.9208569Z -----------stdout-end-------------
2020-08-03T07:03:36.9208761Z  
2020-08-03T07:03:36.9209184Z INFO: Junit data recorded to 'report/TEST-attestation-sgxlkl_attests_to_oe-(debug)-(run-hw)-(8-ethreads)-junit.xml'
2020-08-03T07:03:36.9209862Z Test 'attestation-sgxlkl_attests_to_oe-(debug)-(run-hw)-(8-ethreads)' status - passed (Exit code: '0'; FAILURES in output: '0').
2020-08-03T07:03:36.9210426Z Test 'attestation-sgxlkl_attests_to_oe-(debug)-(run-hw)-(8-ethreads)' status - Completed.
2020-08-03T07:03:36.9210763Z Current Status: passed = 5, failed = 0, disabled = 5, remaining = 34.
2020-08-03T07:03:36.9211059Z Cleaning up /agent/_work/2/s/tests/attestation/sgxlkl_attests_to_oe/report
2020-08-03T07:03:36.9211378Z Changing directory to /agent/_work/2/s
2020-08-03T07:03:36.9211551Z `

For SW: [link](https://dev.azure.com/sgx-lkl/ff25f828-9f87-48e4-94f0-7449609f7e8f/_apis/build/builds/1633/logs/350):
`2020-08-03T06:40:33.9516060Z Changing directory to tests/attestation/sgxlkl_attests_to_oe
2020-08-03T06:40:33.9545553Z tests/attestation/sgxlkl_attests_to_oe/Makefile is disabled for run-sw in Makefile. Skipping test... Remove skip-run-sw from Makefile to enable it.`

hukoyu reviewed (approved):
LGTM

jxyang commented on tests/attestation/sgxlkl_attests_to_oe/oe_enclave/host/host.cpp:
```diff
@@ -0,0 +1,57 @@
+#include <unistd.h>
+#include "tlssrv_u.h"
+
+#define SERVER_PORT "13999"
```
Please use a different port to avoid potential conflict.

triston996 replied:
Any suggestion on the port @jxyang ? So I assume the conflict here is because the Huseyin's test, am I understanding correctly?

triston996 replied:
updated to 17500 @jxyang

triston996 committed:
update port

triston996 commented:
@jxyang I updated the port, also addressed all of your comment. The pipeline breaks because the ltp test, but my test was passed in the pipeline:
hw-mode: https://dev.azure.com/sgx-lkl/sgx-lkl/_build/results?buildId=1802&view=logs&j=b7c9c748-1ca7-5792-6aef-f2abb493d77e&t=21fc6a51-7f72-5416-a625-359f52612cb8&l=9206

sw-mode: https://dev.azure.com/sgx-lkl/sgx-lkl/_build/results?buildId=1802&view=logs&j=4dfb742b-8e24-5ebe-1c3b-99535daead16&t=ba6e03fa-335e-5c86-d35e-1ceb000f4d73&l=2407

please let me know if that looks good to you.

jxyang commented on tests/attestation/sgxlkl_attests_to_oe/oe_enclave/enc/makefile:
```diff
@@ -0,0 +1,42 @@
+	$(CXX) -o tlssrv_enc tlssrv.o gencreds.o tlssrv_t.o $(LDFLAGS) -lmbedtls -lmbedcrypto -loehostsock -loehostresolver -loecore -loelibc -loesyscall
+
+sign:
+	oesign sign -e tlssrv_enc -c enc.conf -k private.pem
```
Since the sgx-lkl enclave doesn't care about the MRSIGNER of oe enclave, we can get rid of the below private.pem file and generate a random private key here.

jxyang replied:
use the following before oesign:
```
openssl genrsa -out private.pem -3 3072
```

triston996 replied:
@jxyang I removed private key and use openssl to generate one. Let me know if that looks good to you

triston996 committed:
use openssl generate private key
